app/scripts/views/settings/clients.js

```diff
@@ -137,33 +136,33 @@ define(function (require, exports, module) {
       }
     },
 
-    _onRefreshClientsList () {
-      if (this.isPanelOpen()) {
-        this.logViewEvent('refresh');
-        // only refresh devices if panel is visible
-        // if panel is hidden there is no point of fetching devices
-        this._fetchAttachedClients().then(() => {
-          this.render();
-        });
-      }
-    },
-
     _onGetApp (event) {
       var appType = this.$el.find(event.currentTarget).data('get-app');
       this.logViewEvent(`get.${appType}`);
     },
 
     openPanel () {
       this.logViewEvent('open');
-      this._fetchAttachedClients();
+      this.$el.find('.clients-refresh').trigger('submit');
```

Comment:
I tested this locally and did not see the progress indicator from your gif example. Tracing through the code, there doesn't appear to be a spinner class of sorts being applied to the button while fetching clients. Is there something I am missing?